app/lib/submission_builder/ty2024/states/md/documents/md502.rb

```diff
@@ -147,6 +151,10 @@ def has_exemptions?
     has_dependent_exemption? || has_line_a_or_b_exemptions
   end
 
+  def has_state_tax_computation?
+    calculated_fields.fetch(:MD502_DEDUCTION_METHOD) == "S" || calculated_fields.fetch(:MD502_LINE_22)&.positive?
```

Comment:
[pebble] what is the significance of the `:MD502_DEDUCTION_METHOD` here?  Not seeing that requirement in the story but I probably am missing something

Looks like `NetIncome`, `ExemptionAmount`, _and_ `TaxableNetIncome` rely on `calculated_fields.fetch(:MD502_DEDUCTION_METHOD) == "S"`, can we make a wrapping `if` block around these instead of checking them over and over?

Reply:
@arinchoi03 there was already an XML element in the `StateTaxComputation` block that would only be created if `calculated_fields.fetch(:MD502_DEDUCTION_METHOD) == "S"`. This method falls in line with the pattern of the other methods that check for the presence of an inner XML element before adding the outer block like `has_exemptions?` or `has_dependent_exemption?`. Maybe I can save it in an instance variable and wrap lines 104 and 105 with a block